File: cpp/08_076.cpp
#include <string>
#include <algorithm> // for max/min

class Solution {
public:
    int evaluateExpression(std::string expr) {
        int index = 0;
        return evaluate(expr, index);
    }
    
private:
    int evaluate(const std::string& expr, int& index) {
        char ch = expr[index++];
        
        // Check if it's an operand (letter)
        if (ch >= 'a' && ch <= 'z') {
            return ch - 'a' + 1;
        }
        
        // It's an operator, evaluate the two operands
        int leftVal = evaluate(expr, index);
        int rightVal = evaluate(expr, index);
        
        // Apply the operator
        switch (ch) {
            case '+': return leftVal + rightVal;
            case '-': return leftVal - rightVal;
            case '*': return leftVal * rightVal;
            case '/': return leftVal / rightVal;
            case '&#x24;': return std::max(leftVal, rightVal);
            case '&': return std::min(leftVal, rightVal);
            default: return 0; // Should not reach here
        }
    }
};

File: python/08_076.py
def evaluate_expression(expr):

    def evaluate(index):
        char = expr[index]
        
        # Check if it's an operand (letter)
        if 'a' <= char <= 'z':
            return ord(char) - ord('a') + 1, index + 1
        
        # It's an operator, evaluate the two operands
        left_val, next_index = evaluate(index + 1)
        right_val, next_index = evaluate(next_index)
        
        # Apply the operator
        if char == '+':
            return left_val + right_val, next_index
        elif char == '-':
            return left_val - right_val, next_index
        elif char == '*':
            return left_val * right_val, next_index
        elif char == '/':
            return left_val // right_val, next_index
        elif char == '&#x24;':
            return max(left_val, right_val), next_index
        elif char == '&':
            return min(left_val, right_val), next_index
        
    result, _ = evaluate(0)
    return result